authorization/src/main/java/com/terabite/movement/controller/CategoryController.java

```diff
@@ -14,33 +15,39 @@
 @CrossOrigin(allowCredentials = "true", origins = "http://localhost:3000")
 @RestController
 @RequestMapping("/v1/cat")
-public class CategoryController
-{
+public class CategoryController {
     private final CategoryRepository categoryRepository;
     private final MovementRepository movementRepository;
 
-    public CategoryController(CategoryRepository categoryRepository, MovementRepository movementRepository)
-    {
+    public CategoryController(CategoryRepository categoryRepository, MovementRepository movementRepository) {
         this.categoryRepository = categoryRepository;
         this.movementRepository = movementRepository;
     }
 
     @GetMapping("/categories")
-    public ResponseEntity<List<Category>> getAllCategories()
-    {
+    public ResponseEntity<List<Category>> getAllCategories() {
         List<Category> categories = new ArrayList<Category>();
 
         categoryRepository.findAll().forEach(categories::add);
 
-        if(categories.isEmpty())
-        {
+        if (categories.isEmpty()) {
             return new ResponseEntity<>(HttpStatus.NO_CONTENT);
         }
 
         return new ResponseEntity<>(categories, HttpStatus.OK);
 
     }
 
+    @GetMapping("/categories/{id}")
+    public ResponseEntity<Category> getCategoryById(@PathVariable("id") long id) {
+        Category category = categoryRepository.findById(id);
+        if (category == null) {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+
+        return new ResponseEntity<>(category, HttpStatus.OK);
+    }
+
     @GetMapping("/movements/{movement_id}/categories")
```

Comment:
This might be better of in movement endpoint like `v1/mov/{id}/category` or `v1/mov/movement/{id}/category`